IPs/RSKIP106.md

```diff
@@ -0,0 +1,135 @@
+
+```
+deriveExtendedPublicKey('tpubD6NzVbkrYhZ4YHQqwWz3Tm1ESZ9AidobeyLG4mEezB6hN8gFFWrcjczyF77Lw3HEs6Rjd2R11BEJ8Y9ptfxx9DFknkdujp58mFMx9H5dc1r', '2/3/4') => 'tpubDCGMkPKredy7oh6zw8f4ExWFdTgQCrAHToF1ytny3gbVy9GkUNK2Nqh7NbKbh8dkd5VtjUiLJPkbEkeg29NVHwxYwzHJFt9SazGLZrrU4Y4'
+```
+
+### extractPublicKeyFromExtendedPublicKey
+
+The method `extractPublicKeyFromExtendedPublicKey(string xpub) returns (bytes)` takes as input an extended public key encoded in base58Check (as described in ref #4) and returns a byte array corresponding to the compressed public key of the given extended public key.
+
+#### Validations
+
+- `xpub` must be a valid base58check-encoded extended public key (either for mainnet or testnet).
+
+#### Sample usage
+
+Given the base58Check-encoded extended public key `tpubDCwK6XsmwUx641qZ6Uyb2pcZXMCoFcyNBFZLbxwJT4iwgmUhY9DsW9FgaFUNE5s1KtM1upxpDuCvx1TbAqQMLMGFF5b1F6KpRpuHcLwcp4k` we would have that:
+
+```
+extractPublicKeyFromExtendedPublicKey('tpubDCwK6XsmwUx641qZ6Uyb2pcZXMCoFcyNBFZLbxwJT4iwgmUhY9DsW9FgaFUNE5s1KtM1upxpDuCvx1TbAqQMLMGFF5b1F6KpRpuHcLwcp4k') => '0x0360a7239100410a4f6019778fce803c6b168a9b1922a0eafbdff11e866070d9f8'
+```
+
+### getMultisigScriptHash
+
+The method `getMultisigScriptHash(uint8 minimumSignatures, bytes[] publicKeys) returns (bytes)` takes as input a minimum required number of signatures and an array of (either compressed or uncompressed) secp256k1 public keys and returns the 20-byte hash (i.e., output of `hash160`) of the Bitcoin N-of-M multisig scriptPub corresponding to the given arguments. It is important to notice that the compressed version of the given public keys will be used to generate the script. Therefore, public key compression will not make a difference in the function output.
+
+#### Validations
+
+- `minimumSignatures` must be greater than zero and lower or equal to the number of public keys.
+- `publicKeys` must be at least 1 and at most 15 (see ref #3 for details on the restriction).
+- Each of the elements of `publicKeys` must be either of size 33 (compressed) or size 65 (uncompressed).
+
+#### Sample usage
+
+Given the following public keys:
+
+- 02276a07b202503d39a43896300224fb649818d1486b0eefde6c9cd7cf1e32eed8
+- 0363e3de0d55459387f221dcda80f9283e303a08bcfeb9cf32875a43fefc7ecb1f
+- 03173019874589385b0295b839bb70e5d3c33178c110bacd872e245b656b3e8f43
+
+and _two_ minimum required signatures, we would have that:
+
+```
+getMultisigScriptHash(2, ['0x02276a07b202503d39a43896300224fb649818d1486b0eefde6c9cd7cf1e32eed8', '0x0363e3de0d55459387f221dcda80f9283e303a08bcfeb9cf32875a43fefc7ecb1f', '0x03173019874589385b0295b839bb70e5d3c33178c110bacd872e245b656b3e8f43']) => '0xafa22d52922aa6657417c55b6cebdf708b594576'
+```
+
+_Note_: The public keys used here are compressed, but might as well be uncompressed. The output should be exactly the same since the function handles the conversion.
+
+## Rationale
+
+TODO (if any)
```

Comment:
Add the gas consumption for each method somewhere.

Reply:
Perfect. Will do when we get the final estimations in.

Reply:
I've added in the gas consumptions.